### ISSUE TITLE
Version 9 incompatible with relay-compiler version 7?

### ISSUE THREAD

**Issue by m64253** [relay-tools/relay-compiler-language-typescript#151]
I guess this commit to relay-compiler https://github.com/facebook/relay/commit/536f7815de68a0b43f11b65a16c32d3a2a42b8c7 are causing some deep imports like this to fail? https://github.com/relay-tools/relay-compiler-language-typescript/blob/bbaf3007e175f6141aa3d9f4100010081a876c0d/src/TypeScriptGenerator.ts#L16

**Comment by smmoosavi** [comment]
`@types/relay-compiler` needs to be updated and add types for new files

**Comment by smmoosavi** [comment]
after fixing file imports, this error will occurs

```
ERROR:
  Error writing modules:
  Error: Invalid AST Node: { QUERY_TYPE_KEY: Symbol(Query), MUTATION_TYPE_KEY: Symbol(Mutation), SUBSCRIPTION_TYPE_KEY: Symbol(Subsription), _baseSchema: { __validationErrors: [], extensions: undefined, astNode: [Object], extensionASTNodes: undefined, __allowedLegacyNames: [], _queryType: Root, _mutationType: null, _subscriptionType: null, _directives: [Array], _typeMap: [Object], _possibleTypeMap: {}, _implementations: [Object] }, _extendedSchema: { __validationErrors: undefined, extensions: undefined, astNode: [Object], extensionASTNodes: [], __allowedLegacyNames: [], _queryType: Root, _mutationType: null, _subscriptionType: null, _directives: [Array], _typeMap: [Object], _possibleTypeMap: {}, _implementations: [Object] }, _typeMap: {}, _fieldsMap: {}, _typeNameMap: {}, _clientIdMap: {}, _possibleTypesMap: {}, _directivesMap: {} }
at visit (/path-to-project/node_modules/graphql/language/visitor.js:236:15)
at Object.visitIR [as visit] (/path-to-project/node_modules/relay-compiler/lib/core/GraphQLIRVisitor.js:41:10)
at Object.exports.generate (/path-to-project/node_modules/relay-compiler-language-typescript/lib/TypeScriptGenerator.js:68:46)
at /path-to-project/node_modules/relay-compiler/bin/relay-compiler:11541:64
at Generator.next (<anonymous>)
at asyncGeneratorStep (/path-to-project/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
at _next (/path-to-project/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
at /path-to-project/node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
at new Promise (<anonymous>)
at /path-to-project/node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12
```

the error throws from this line
```ts
const ast: ts.Statement[] = IRVisitor.visit(node, createVisitor(options));
```

**Comment by janicduplessis** [comment]
Relay now uses its own ast so some updates are needed to use their schema helpers instead of using graphql-js.

I made the changes to the compiled files quickly just to get it working if that can help someone https://gist.github.com/janicduplessis/6c134b4c21168aa28b81e3fd818ec9bd

We'll need to update this based on the recent changes in https://github.com/facebook/relay/blob/master/packages/relay-compiler/language/javascript/RelayFlowGenerator.js and https://github.com/facebook/relay/blob/master/packages/relay-compiler/language/javascript/RelayFlowTypeTransformers.js

**Comment by sorenhoyer** [comment]
I guess it'll be quite a lot of work to get it relay v7 compatible. I made a fork with some of the changes here: https://github.com/sorenhoyer/relay-compiler-language-typescript - not sure I'll have time over the weekend to continue, so hopefully somebody else is working on this and will beat me to it :D

**Comment by sorenhoyer** [comment]
Started some work on it here in a draft PR. https://github.com/relay-tools/relay-compiler-language-typescript/pull/154

**Comment by alloy** [comment]
Thanks for all jumping on this, people, really happy with the community effort 🙌

**Comment by renanmav** [comment]
Fixed by #155
